test/e2e/csv_e2e_test.go

```diff
@@ -316,6 +372,69 @@ func TestCreateCSVRequirementsMet(t *testing.T) {
 	compareResources(t, fetchedCSV, sameCSV)
 }
 
+func TestCreateCSVRequirementsMetAPIService(t *testing.T) {
+	defer cleaner.NotifyTestComplete(t, true)
+
+	c := newKubeClient(t)
+	crc := newCRClient(t)
+
+	strategy := install.StrategyDetailsDeployment{
+		DeploymentSpecs: []install.StrategyDeploymentSpec{
+			{
+				Name: genName("dep-"),
+				Spec: newNginxDeployment(genName("nginx-")),
+			},
+		},
+	}
+	strategyRaw, err := json.Marshal(strategy)
+	require.NoError(t, err)
+
+	csv := v1alpha1.ClusterServiceVersion{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       v1alpha1.ClusterServiceVersionKind,
+			APIVersion: v1alpha1.ClusterServiceVersionAPIVersion,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: genName("csv"),
+		},
+		Spec: v1alpha1.ClusterServiceVersionSpec{
+			InstallStrategy: v1alpha1.NamedInstallStrategy{
```

Comment:
You could also use [this helper](https://github.com/operator-framework/operator-lifecycle-manager/blob/678fea44cfea340f798b8aa629aa8f388a32b66f/test/e2e/installplan_e2e_test.go#L86).